tools/backend/rtl-constant-generator-verilog/rtl-constant-generator-verilog.cpp

```diff
@@ -0,0 +1,93 @@
+//===- rtl-text-generator.cpp - Text-based RTL generator --------*- C++ -*-===//
+//
+// Dynamatic is under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Simple generator for RTL components, which takes as input an RTL file,
+// replaces user-provided strings (with regex support) within it, and dumps the
+// result at a specified location.
+//
+//===----------------------------------------------------------------------===//
+
+#include "dynamatic/Support/RTL.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/InitLLVM.h"
+#include "llvm/Support/SourceMgr.h"
+#include <fstream>
+#include <map>
+
+using namespace llvm;
+
+static cl::OptionCategory mainCategory("Tool options");
+
+static cl::opt<std::string> inputRTLPath(cl::Positional, cl::Required,
+                                         cl::desc("<input file>"),
+                                         cl::cat(mainCategory));
+
+static cl::opt<std::string> outputRTLPath(cl::Positional, cl::Required,
+                                          cl::desc("<output file>"),
+                                          cl::cat(mainCategory));
+
+static cl::list<std::string> replacements(
+    cl::Positional, cl::ZeroOrMore,
+    cl::desc(
+        "<text replacements, two-by-two (text to replace, then replacement)>"),
+    cl::cat(mainCategory));
```

Comment:
Since this generator is just for the Verilog constant you don't actually need to pass the strings to replace as command-line argument since you statically know what they are. Directly pass the two replacement values on the command line and associate them to the known strings they should replace directly in the `main` before calling `replaceRegexes`. This allows you to get rid of the special-cased for loop.